### PR TITLE
doc: matter: add QR codes to samples

### PR DESCRIPTION
**Pull request by greg-fer**
Added QR code image and data to Matter samples.
Applied other minor fixes.
KRKNWK-15935.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: applications/matter_weather_station/README.rst [hunk 1]
```diff
@@ -264,6 +262,51 @@ After programming the application, perform the following steps to test the Matte
 
          Relative humidity measurement type selection
 
+Onboarding information
+----------------------
+
+When you start the commissioning procedure, the controller must get the onboarding information from the Matter accessory device.
+The onboarding information depends on your commissioner setup.
```
Comment (Damian-Nordic) [inline applications/matter_weather_station/README.rst:269]:
```suggestion
The onboarding information representation depends on your commissioner setup.
```

File: applications/matter_weather_station/README.rst [hunk 2]
```diff
@@ -264,6 +262,51 @@ After programming the application, perform the following steps to test the Matte
...
+
+For this application, the data payload, which includes the device discriminator and setup PIN code, is encoded and shared using an NFC tag.
+When using the debug configuration, you can also get this type of information from the USB interface logs.
+
+Alternatively, depending on your build type, you can also use one of the following :ref:`onboarding information formats <ug_matter_network_topologies_commissioning_onboarding_formats>` to provide the commissioner with the data required:
+
+* For the default debug build type:
```
Comment (Damian-Nordic) [inline applications/matter_weather_station/README.rst:276]:
```suggestion
* For the default debug and release build type:
```

File: doc/nrf/ug_matter_overview_commissioning.rst [hunk 3]
```diff
@@ -29,11 +29,18 @@ The data payload includes the following information, among other things:
...
-* Within Manual Pairing Code.
-* Within QR Code.
+Onboarding information formats
+==============================
+
+The data payload can be encoded in one of the following ways, either included on-device or in the packaging for the end products (or both):
+
+* Manual Pairing Code - which provides the onboarding information as a sequence of digits that can be used with most Matter commissioners.
+* QR Code - which you can scan using the appropriate mobile application for your ecosystem.
   For testing purposes, this code can be printed to the UART console or be shared using an NFC tag.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_overview_commissioning.rst:41]: This actually applies to QR Code Payload.

File: doc/nrf/ug_matter_overview_commissioning.rst [hunk 4]
```diff
@@ -29,11 +29,18 @@ The data payload includes the following information, among other things:
...
+* QR Code Payload - which is a code made of digits and letters that you can use in command-line tools.
```
Comment (Damian-Nordic) [inline doc/nrf/ug_matter_overview_commissioning.rst:42]: Just throwing a shorter term, but it's up to you which version you prefer.
```suggestion
* QR Code Payload - which is an alphanumeric code that you can use in command-line tools.
```

**Review by Damian-Nordic — APPROVED** [review]
Looks great. Just a couple of nits from me.

**Review by markaj-nordic — APPROVED** [review]

**Pull request merged**